### PR TITLE
docs(guide/displaying-data): proofread

### PR DESCRIPTION
**Pull request by chalin**
- TS prose updated to include @kwalrath's revisions from a while ago, with some of my edits as well.
- Dart prose simplified by removing discussion of "additions to `pubspec.yaml`" which are no longer necessary given the current state of QuickStart.
- E2e suites passed:
  - public/docs/_examples/displaying-data/dart
  - public/docs/_examples/displaying-data/ts

Contributes to #1598 and #1508.
Depends on #1818.

### CODE REVIEW

**Comment by chalin** [comment]
@Foxandxss @wardbell @kwalrath : ready for review. As usual, it is best viewed by ignoring whitespace diffs.

**Comment by Foxandxss** [comment]
TS part works for me.

File: public/docs/ts/latest/guide/displaying-data.jade [hunk 1]
```diff
@@ -179,65 +187,70 @@ figure.image-display
...
-  Consider the first parameter:
-+makeExample('displaying-data/ts/app/hero.ts', 'id-parameter')
+    Consider the first parameter:
 
-:marked
-  That brief syntax does a lot:
-  * declares a constructor parameter and its type
-  * declares a public property of the same name
-  * initializes that property with the corresponding argument when we "new" an instance of the class
+  +makeExcerpt('app/hero.ts ()', 'id')
```
Comment (Foxandxss) [inline public/docs/ts/latest/guide/displaying-data.jade]: What does the ` ()` in here?

Comment (chalin) [inline public/docs/ts/latest/guide/displaying-data.jade]: Because of the `()` the mixin will render the title as `app/hero.ts (excerpt)`.

File: public/docs/ts/latest/guide/displaying-data.jade [hunk 2]
```diff
@@ -1,34 +1,35 @@
...
 :marked
   # Table Of Contents
-  
+
   * [Showing component properties with interpolation](#interpolation)
-  * [Showing an array property with NgFor](#ngFor)
+  * [Showing !{_an} !{_array} property with NgFor](#ngFor)
   * [Conditional display with NgIf](#ngIf)
-  
+  <li if-docs="ts">only ts</li>
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/displaying-data.jade]: Why is this line here? This shows up as a weird "only ts" bullet item in the TS version of the docs...

File: public/docs/ts/latest/guide/displaying-data.jade [hunk 3]
```diff
@@ -179,65 +187,70 @@ figure.image-display
...
+  Let's make the `heroes` property in our component return !{_an} !{_array} of these `Hero` objects.
+
++makeExcerpt('app/app.component.3.ts', 'heroes')
 
 :marked
   We'll have to update the template.
-  At the moment it displays the entire `hero` object, which used to be a string value.
-  Let's fix that so we interpolate the `hero.name` property.
-+makeExample('displaying-data/ts/app/app.component.3.ts', 'template','app.component.ts (template)')(format=".")
+  At the moment it displays the `Hero` object.
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/displaying-data.jade]: `Hero` object doesn't seem quite right here. Revert back to "the string value of the `Hero` object"?

**Comment by kwalrath** [comment]
I did a side-by-side comparison, and apart from the couple of minor things I commented on, LGTM.

File: public/docs/ts/latest/guide/displaying-data.jade (hunk 2 above)
Comment (chalin) [inline public/docs/ts/latest/guide/displaying-data.jade]: That was just for a test. It should not have been left in (no sure how it slipped by). Thanks for catching it!

Fixed.

File: public/docs/ts/latest/guide/displaying-data.jade (hunk 3 above)
Comment (chalin) [inline public/docs/ts/latest/guide/displaying-data.jade]: How about
> At the moment it displays the `Hero` object's attributes:`

**Comment by chalin** [comment]
Fixed and committed.

**Comment by kwalrath** [comment]
LGTM
